x-pack/plugins/security_solution/public/detections/components/rules/description_step/index.tsx

```diff
@@ -204,12 +205,29 @@ export const getDescriptionItem = (
   } else if (field === 'responseActions') {
     return [];
   } else if (field === 'groupByFields') {
+    // only query rule can have suppression group by fields
+    const ruleType: Type = get('ruleType', data);
+    if (!isQueryRule(ruleType)) {
+      return [];
+    }
     const values: string[] = get(field, data);
     return buildAlertSuppressionDescription(label, values);
   } else if (field === 'groupByRadioSelection') {
     return [];
   } else if (field === 'groupByDuration') {
-    if (get('groupByFields', data).length > 0) {
+    const ruleType: Type = get('ruleType', data);
+    // only query and threshold rules can have duration property
+    if (!isQueryRule(ruleType) && !isThresholdRule(ruleType)) {
+      return [];
+    }
+
+    // threshold rule can have suppression duration without grouping fields, but should be explicitly enabled by user
```

Comment:
Nit: this comment is basically explaining the purpose of `enableThresholdSuppression`, which is mostly redundant with how it's used below; I think it could be deleted without losing much.

Reply:
thanks for suggestions on comments. I applied most of them, but think this one is a bit more complex, so tend to leave it as it is